server/controllers/user-books-controller.js

```diff
@@ -0,0 +1,351 @@
+import {
+  UserDetails,
+  Books,
+  BookLendings,
+} from '../models';
+
+class userBookInteraction {
+  static validateEntry(
+    userId, bookId, lendId,
+  ) {
+    return new Promise((resolve, reject) => {
+      if (isNaN(userId)) {
+        reject('Invalid User id');
+      } else if (isNaN(bookId)) {
+        reject('Invalid Book id');
+      } else if (isNaN(lendId)) {
+        reject('Invalid');
+      } else {
+        resolve('Valid');
+      }
+    });
```

Comment:
I believe you should abstract some re-useable code block to a function and call it in your controller. Your controllers are pretty long.